frontend/src/features/admin-form/preview/PreviewFormProvider.tsx

```diff
@@ -0,0 +1,214 @@
+    isNotFormId,
+  } = useCommonFormProvider(formId)
+
+  useEffect(() => {
+    if (data) {
+      if (!cachedDto) {
+        trackVisitPublicForm(data.form)
+        setCachedDto(data)
+      } else if (!desyncToastIdRef.current && !isEqual(data, cachedDto)) {
+        desyncToastIdRef.current = toast({
+          status: 'warning',
+          title: (
+            <Text textStyle="subhead-1">
+              The form has been modified and your submission may fail.
+            </Text>
+          ),
+          description: (
+            <Text as="span">
+              <Link href="">Refresh</Link> for the latest version of the form.
+            </Text>
+          ),
+          duration: null,
+        })
+      }
+    }
+  }, [data, cachedDto, toast, desyncToastIdRef])
+
+  const getTransactionId = useCallback(async () => {
+    if (!vfnTransaction || isPast(vfnTransaction.expireAt)) {
+      const result = await createTransactionMutation.mutateAsync()
+      setVfnTransaction(result)
+      return result.transactionId
+    }
+    return vfnTransaction.transactionId
+  }, [createTransactionMutation, vfnTransaction])
+
+  const isFormNotFound = useMemo(() => {
+    return (
+      error instanceof HttpError && (error.code === 404 || error.code === 410)
+    )
+  }, [error])
+
+  const expiryInMs = useMemo(() => {
+    if (!vfnTransaction?.expireAt) return null
+    return differenceInMilliseconds(vfnTransaction.expireAt, Date.now())
+  }, [vfnTransaction])
+
+  const generateVfnExpiryToast = useCallback(() => {
+    if (vfnToastIdRef.current) {
+      toast.close(vfnToastIdRef.current)
+    }
+    const numVerifiable = cachedDto?.form.form_fields.filter((ff) =>
+      get(ff, 'isVerifiable'),
+    ).length
+
+    if (numVerifiable) {
+      vfnToastIdRef.current = toast({
+        duration: null,
+        status: 'warning',
+        isClosable: true,
+        description: simplur`Your verified field[|s] ${[
+          numVerifiable,
+        ]} [has|have] expired. Please verify [the|those] ${[
+          numVerifiable,
+        ]} field[|s] again.`,
+      })
+    }
+  }, [cachedDto?.form.form_fields, toast, vfnToastIdRef])
+
+  useTimeout(generateVfnExpiryToast, expiryInMs)
+
+  const showErrorToast = useCallback(() => {
+    toast({
+      status: 'danger',
+      description:
+        'An error occurred whilst processing your submission. Please refresh and try again.',
+    })
+  }, [toast])
+
+  const handleSubmitForm = useCallback(async () => {
+    const { form } = cachedDto ?? {}
+    if (!form) return
+
+    let captchaResponse: string | null
+    try {
+      captchaResponse = await getCaptchaResponse()
+    } catch (error) {
+      if (error instanceof RecaptchaClosedError) {
+        // Do nothing if recaptcha is closed.
+        return
+      }
+      return showErrorToast()
+    }
+
+    setSubmissionData({ id: 'submissionId', timeInEpochMs: Date.now() })
+  }, [cachedDto, getCaptchaResponse, showErrorToast])
```

Comment:
ah, for context, we actually have an endpoint for admins to preview and test their form submissions. It still goes to our servers and triggers the entire pipeline (with the exception of actually storing the response metadata). 

Can check it out [here](https://github.com/opengovsg/FormSG/blob/form-v2/develop/src/app/routes/api/v3/admin/forms/admin-forms.preview.routes.ts#L38) 

(so we still need mutations here, the only difference between preview and public form is the GET and POST endpoints. That's why we can more or less use all the methods except the queries, and maybe some UI differences)

Reply:
thank u! changed to use those endpoints :)